backend/app/controllers/spree/admin/resource_controller.rb

```diff
@@ -134,9 +134,10 @@ def belongs_to(model_name, options = {})
     end
   end
 
-  def resource_not_found
-    flash[:error] = flash_message_for(model_class.new, :not_found)
-    redirect_to collection_url
+  def resource_not_found(_err, options = {})
+    options[:flash_class] ||= model_class
+    options[:redirect_url] ||= collection_url
+    super(options)
```

Comment:
Since you're not using the error, you can remove that argument, continue using keyword arguments, add default values, and rely on the fact that super passes through the arguments implicitly:

```suggestion
  def resource_not_found(flash_class: model_class, redirect_url: collection_url)
    super
```

Reply:
This looks so much neater, thank you. I was so stuck on passing arguments into super.

If I don't have the _err argument then for almost all previously working redirects, including the images, prices and product_properties controllers I am getting an 
`ArgumentError at /admin/products/ruby-asdf/product_properties
wrong number of arguments (given 1, expected 0)`

I believe that this is due to an error message is being passed to the #resource_not_found method by the default rescue method.
 
Is there a more elegant solution than putting `_err = nil` in the argument list for both the base and resource controllers? The other option I imagine is a `*` as the first argument in the methods for both controllers.